src/effects/VST/VSTEffect.h

```diff
@@ -218,6 +222,34 @@ struct VSTEffectWrapper : public VSTEffectLink, public XMLTagHandler
    virtual int GetProcessLevel() = 0;
    virtual void SizeWindow(int w, int h) = 0;
    virtual void Automate(int index, float value) = 0;
+
+   bool Load();
+   PluginPath   mPath;
+
+   // Define a manager class for a handle to a module
+#if defined(__WXMSW__)
+   using ModuleHandle = std::unique_ptr<wxDynamicLibrary>;
+#else
+   struct ModuleDeleter {
+      void operator() (void*) const;
+   };
+   using ModuleHandle = std::unique_ptr < char, ModuleDeleter >;
+#endif
+
+   ModuleHandle mModule{};
+
+   wxString mVendor;
+   wxString mDescription;
+   int      mVersion;
+   bool     mInteractive{ false };
+   unsigned mAudioIns{ 0 };
+   unsigned mAudioOuts{ 0 };
+   int      mMidiIns{ 0 };
+   int      mMidiOuts{ 0 };
+   bool     mAutomatable;
+
+   virtual void callSetProgram(int index) = 0;
+   virtual void Unload() = 0;
```

Comment:
I see you add two virtual functions to the wrapper when you moved Load() into it.

I understand virtual `Unload` lets wrapper avoid mention of `mDialog`.

But I think the last part of `Unload`:

```
   if (mModule)
   {
#if defined(__WXMAC__)
      mResource.reset();
      mBundleRef.reset();
#endif

      mModule.reset();
      mAEffect = NULL;
   }
```

should be the responsibility of Wrapper, balancing what it does in load.  Make those lines a non-virtual function; call it in the destructor of LV2Wrapper, and also on the failure path of `Load()`.

And I think you defined virtual `callSetProgram` to avoid dependending on `mSlaves`.  But `Load()` only needs `callSetProgram` to initialize a plug-in or an instance which will not yet have any slaves.  So I think you don't need this function, and can instead let `Load` call `callSetProgramB`

Reply:
ok, done